roles/preflight_checks/action_plugins/validate_inventory.py

```diff
@@ -109,6 +109,28 @@ def check_ansible_user_is_not_root(task_vars):
             )
 
 
+def check_no_capital_letter_in_hostnames(task_vars):
+    '''Ensure there is only lower case hostnames in the inventory
+
+    Kubespray set the hostname with lower case based on the inventory content
+    but this breaks calico, therefore we should enforce the usage of lower
+    letter only in the hostnames defined in the inventory
+    Kubnespray uses the `inventory_hostname` variable
+    '''
+
+    for host, hostvars in sorted(task_vars['hostvars'].items()):
+        if any([letter.isupper() for letter in host]):
```

Comment:
I think we should extend this and make sure there are no other nonsense characters in the name either. Basically `^[a-z0-9]([a-z0-9\-]*[a-z0-9])?$` or something along those lines ('Starts with a-z0-9, can have a-z0-9 or - in the middle, ends with a-z0-9')

Reply:
Hum, what about unicode characters then ? Technically it's valid, but does calico supports that ? If we go that far, shouldn't we do a full validation of the hostname provided as being valid FQDN instead ?

Reply:
I would rule out too 'particular' things for now.